src/FSharp.Data.GraphQL.Server/Planning.fs

```diff
@@ -335,44 +360,37 @@ let internal planOperation (ctx: PlanningContext) : ExecutionPlan =
         Include = incl
         IsNullable = false }
     let resolvedInfo = planSelection ctx ctx.Operation.SelectionSet rootInfo (ref [])
-    let topFields =
+    let fields =
         match resolvedInfo.Kind with
         | SelectFields tf -> tf
-        | x -> failwith <| sprintf "Expected SelectFields Kind, but got %A" x
+        | x -> failwith $"Expected SelectFields Kind, but got %A{x}"
     let variables = planVariables ctx.Schema ctx.Operation
-    match ctx.Operation.OperationType with
-    | Query ->
+
+    let executionPlan =
         { DocumentId = ctx.DocumentId
           Operation = ctx.Operation
-          Fields = topFields
           RootDef = ctx.Schema.Query
-          Strategy = Parallel
+          Fields = fields
           Variables = variables
-          Metadata = ctx.Metadata
-          ValidationResult = ctx.ValidationResult }
+          Strategy = Parallel
+          Metadata = ctx.Metadata }
+    match ctx.Operation.OperationType with
+    | Query -> executionPlan
```

Comment:
I don't think we should do this as it adds an unnecessary allocation

Reply:
True.  I was doing it more for clarity, since it makes the differences between the returned structures more obvious in the code.  I don't mind reverting it though.

Reply:
Reverted in https://github.com/Lula-Technologies-Inc/FSharp.Data.GraphQL/pull/9/commits/1c9745bc9380e77d98f8f00bccc0f22b50460b6f